cpp/include/cudf/io/json.hpp

```diff
@@ -340,6 +349,13 @@ class json_reader_options {
    */
   void set_byte_range_size(size_type size) { _byte_range_size = size; }
 
+  /**
+   * @brief Set delimiter separating records in JSON lines
+   *
+   * @param delimiter Delimiter separating records in JSON lines
+   */
+  void set_delimiter(char delimiter) { _delimiter = delimiter; }
```

Comment:
Maybe a bit paranoid, but could we verify here that passed delimiter is viable?

Reply:
Good question, it is a difficult problem to verify that the delimiter is not present anywhere else in the input JSON string i.e in non-linebreak positions. If it is present, the parsing would be incorrect i.e. if the delimiter collides with another input symbol, we don't throw an error.
It is a difficult problem because we would first need to identify the context (root, list, struct) in which the delimiter character occurs, and then error out if we see it somewhere other than end of line.

Reply:
I assume it's more about rejecting the obviously wrong choices, like brackets, braces, quotes, etc. At least that's what I would consider here.

Reply:
Ah okay, thank you for the suggestion. I'll add checks for `'{', '[', ']', '}', ',', ':', '"', '\'', '\\', ' ', '\t', '\r'`